File: core/src/main/java/heast/core/network/packets/s2c/login/SuccessS2CPacket.java
package heast.core.network.packets.s2c.login;

import heast.core.network.Packet;
import heast.core.network.PacketBuf;
import heast.core.network.listeners.ClientLoginListener;

public final class SuccessS2CPacket implements Packet<ClientLoginListener> {

    public SuccessS2CPacket() {

    }

    public SuccessS2CPacket(PacketBuf buf) {

    }

    @Override
    public void write(PacketBuf buf) {

    }

    @Override
    public void apply(ClientLoginListener listener) {
        listener.onSuccess();
    }
}

File: Core/Network/Packets/S2C/SuccessS2CPacket.cs
using Core.Network.Listeners;

namespace Core.Network.Packets.S2C;

public class SuccessS2CPacket : AbstractPacket
{
    public SuccessS2CPacket()
    {
    }

    public SuccessS2CPacket(PacketBuf buf)
    {
    }

    public override void Write(PacketBuf buf)
    {
    }

    public override void Apply(IPacketListener listener)
    {
        if (listener is IClientHandshakeListener handshakeListener)
        {
            handshakeListener.OnSuccess(this);
        }
    }
}